arch/x86/x86_64/include/uk/asm/tls.h

```diff
@@ -67,5 +81,8 @@ static inline void ukarch_tls_area_copy(void *tls_area)
 
 static inline void *ukarch_tls_pointer(void *tls_area)
 {
-	return tls_area + (_tls_end - _tls_start);
+	__sz tls_len = _tls_end - _tls_start;
+	__sz padding = tls_len % ukarch_tls_area_align();
+
+	return tls_area + padding + tls_len;
```

Comment:
I believe it should have been `return tls_area + ukarch_tls_area_size();`. Otherwise `fs:offset` seems to pick wrong values.

Reply:
`ukarch_tls_area_size()` returns the full TLS area size, including **8 bytes for the TLS pointer itself** appended at the end of it. It does seem to me the TLS pointer address should be `tls_area + ukarch_tls_area_size() - 8`, which is the same as `tls_area + padding + tls_len`. It was already the case before my changes, which amounted to add the padding both to `ukarch_tls_area_size()` and to `ukarch_tls_pointer()`.
Can you provide a [MWE](https://www.wikiwand.com/en/Minimal_working_example) that triggers wrong values for `fs::offset`, so I can have a look? It seems to work on my setup.
That being said, I think returning `tls_area + ukarch_tls_area_size() - 8` would be clearer and less redundant than what is currently present. I'll update the PR with that version.

Reply:
I found some issue on my own, I will investigate and come back to you.

Reply:
I have provided [here](https://github.com/cs-pub-ro/app-helloworld/tree/TLS-alignment) the example which fails on my end. With this commit, the output is `0x0.0x0.0x0`, instead of `0x22.0x4.0x4`.

Reply:
There is some problem unrelated to this patch, that also occurs with current staging. I created a new issue: #290. Let's revisit the alignment issue once #290 is fixed.